### PR TITLE
[stable/postgresql] Allows overwrite default config file using include_dir

### LINKED ISSUE #8916: [stable/postgresql] Read postgresql.conf / pg_hba.conf overrides from values
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Feature request

[stable/postgresql](https://github.com/helm/charts/tree/master/stable/postgresql) chart reads complete `postgresql.conf` and `pg_hba.conf` from files. This is inflexible. I propose to change this chart to:

1. Read configuration overrides from values instead of files (see [stable/rabbitmq](https://github.com/helm/charts/blob/master/stable/rabbitmq/values.yaml#L99) for example);
2. Allow extending `postgresql.conf` instead of overriding it (cf. bitnami/bitnami-docker-postgresql#90).

### PR DESCRIPTION
**Pull request by carrodher**
Signed-off-by: Carlos Rodriguez Hernandez <crhernandez@bitnami.com>

#### What this PR does / why we need it:

Allow extending `postgresql.conf` instead of overriding it. Users can add different files as configMap

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #8916

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

**Comment by carrodher** [comment]
/hold

Waiting until the changes in the image were released

File: stable/postgresql/templates/extended-config-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+kind: ConfigMap
+metadata:
+  name: {{ template "postgresql.fullname" . }}-extended-configuration
+  labels:
+    app: {{ template "postgresql.name" . }}
+    chart: {{ template "postgresql.chart" . }}
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+data:
+{{- if (.Files.Glob "files/conf.d/*.conf") }}
```
Comment (juan131) [inline stable/postgresql/templates/extended-config-configmap.yaml]: You're already doing this check at the beginning of the file

**Review by juan131 — CHANGES_REQUESTED** [review]

File: stable/postgresql/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: postgresql
-version: 3.2.1
+version: 3.2.2
```
Comment (tompizmor) [inline stable/postgresql/Chart.yaml]: you should change the minor version (`3.3.0`)

**Review by juan131 — APPROVED** [review]
/ok-to-test

**Comment by carrodher** [comment]
`10.6.0` tag is released: https://hub.docker.com/r/bitnami/postgresql/tags/

@tompizmor @juan131 can you continue with the review?

**Pull request merged**